library/ssl_tls13_server.c

```diff
@@ -1807,10 +1812,271 @@ static int ssl_tls13_handshake_wrapup( mbedtls_ssl_context *ssl )
     MBEDTLS_SSL_DEBUG_MSG( 2, ( "handshake: done" ) );
 
     mbedtls_ssl_tls13_handshake_wrapup( ssl );
+
+#if defined(MBEDTLS_SSL_SESSION_TICKETS)
+    mbedtls_ssl_handshake_set_state( ssl,
+                                     MBEDTLS_SSL_NEW_SESSION_TICKET );
+#else /* MBEDTLS_SSL_SESSION_TICKETS */
     mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_HANDSHAKE_OVER );
+#endif /* !MBEDTLS_SSL_SESSION_TICKETS */
+    return( 0 );
+}
+
+/*
+ * Handler for MBEDTLS_SSL_NEW_SESSION_TICKET
+ */
+#define SSL_NEW_SESSION_TICKET_SKIP  0
+#define SSL_NEW_SESSION_TICKET_WRITE 1
+MBEDTLS_CHECK_RETURN_CRITICAL
+static int ssl_tls13_write_new_session_ticket_coordinate( mbedtls_ssl_context *ssl )
```

Comment:
I don't really understand this here.

`ssl_tls13_write_new_session_ticket_coordinate()` is called within `MBEDTLS_SSL_PROC_CHK_NEG()` and is marked `MBEDTLS_CHECK_RETURN_CRITICAL` but can't return anything other than 0 or 1.

Is there a lot more to come in here in a future PR?

Reply:
Not a lot more to come here I think and probably no possible failure but I see this as the signature of `coordinate` functions in TLS 1.3 code. This one does not and probably will not return an error but for example the one for the certificate request state in ssl_tls13_client.c (function ssl_tls13_certificate_request_coordinate()) does. Thus I am rather in favor of keeping it like this.

Reply:
okay, that makes sense